org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/themes/IThemeBehaviour.java

```diff
@@ -0,0 +1,27 @@
+package org.eclipse.tm4e.ui.themes;
+
+/**
+ * API to identify dark or light color scheme applied.
+ * 
+ * Motivation: get the opportunity to influence the applying TM themes behavior depending on the eclipse theme.
+ * 
+ * @author Tim Kovalyov
+ *
+ */
+public interface IThemeBehaviour {

```

Comment:
The world of themes is not so binary, it's not dark vs light. It can be blue, yellow, grey... I'm not convinced an API which as such a binary interpretation of theming will scale with time and use-cases.
Why instead of invoking an API to retrieve the theme, TM4E should look at the default background color of the widget and decide whether it's dark or not? Such more dynamic behaviour could prevent from introducing an API just to know whether current theme is dark or not.